script.js

```diff
@@ -1,4 +1,165 @@
+//      - refactor outputMessages to interact with each player 1 and 2
+//      - write logic for player 1 to go first then player 2, then proceed to compare score
+// ver 3. implement comparing dice scores and declare winner
+// ver 4. reset game so that players can play again without refreshing browser page
+
+// === code start ===
+// global variables
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+// helper function
+var rollDice = function () {
+  console.log("Control Flow: Start of rollDice()");
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  console.log("rollDice output, randomInteger: " + randomInteger);
+  return randomInteger;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control Flow: Start of rollDiceForPlayer()");
+  for (let i = 0; i < 2; i += 1) {
+    currentPlayerRolls.push(rollDice());
+  }
+  console.log(
+    "rollDiceForPlayer changes, currentPlayerRolls: " + currentPlayerRolls
+  );
+  return (
+    "Welcome, Player " +
+    currentPlayer +
+    "!<br><br>You rolled: <br>Dice 1: " +
+    currentPlayerRolls[0] +
+    " | Dice 2: " +
+    currentPlayerRolls[1] +
+    ".<br><br>Sweet! Now input either '1' or '2' to choose the corresponding dice to be used as the first digit of your final value."
+  );
+};
+
+var getPlayerScore = function (playerInput) {
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(
+      "Control Flow: Input validation, invalid input... INPUT NOT 1 OR 2"
+    );
+    return (
+      "Invalid input! Please only input '1' or '2' to choose which dice to use as the first digit.<br><br>Your dice rolls are:<br>Dice 1: " +
+      currentPlayerRolls[0] +
+      " | Dice 2: " +
+      currentPlayerRolls[1] +
+      "."
+    );
+  }
+  // input == 1
+  if (playerInput == 1) {
+    console.log("Control Flow: input == 1");
+    var playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  // input == 2
+  if (playerInput == 2) {
+    console.log("Control Flow: input == 2");
+    var playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+  // store playerScore in array
+  allPlayersScore.push(playerScore);
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return (
+    "Player " + currentPlayer + ", your chosen value is: '" + playerScore + "'."
+  );
+};
+
+var comparePlayerScores = function () {
+  compareMessage =
+    "Player 1 Score: " +
+    allPlayersScore[0] +
+    "<br>Player 2 Score: " +
+    allPlayersScore[1];
+  // player 1 wins
+  if (allPlayersScore[0] > allPlayersScore[1]) {
```

Comment:
It would be more ideal to use if, else if and else here even though technically it doesn't make any difference.